phylokit/distance.py

```diff
@@ -122,3 +122,66 @@ def kc_distance(ds1, ds2, lambda_=0.0):
     )
 
     return np.linalg.norm((1 - lambda_) * (m[0] - m[1]) + lambda_ * (M[0] - M[1]))
+
+
+def _get_encoded_tree(postorder, left_child, right_sib):
+    tree_sets = [set()] * len(left_child)
+    for u in postorder:
+        v = left_child[u]
+        num_children = 0
```

Comment:
Can simplify this a bit:
```python3
v = left_child[u]
if v == -1:
     tree_sets[u] = {u}
else:
     while v != -1:
            tree_sets[u] |= tree_sets[v]
            v = right_sib[v]
```

Reply:
```
[set(), set(), set(), set(), set(), set(), set()]
[{0}, set(), set(), set(), set(), set(), set()]
[{0}, {1}, set(), set(), set(), set(), set()]
[{0}, {1}, {0, 1}, {0, 1}, {0, 1}, {0, 1}, {0, 1}]
[{0}, {1}, {2}, {0, 1}, {0, 1}, {0, 1}, {0, 1}]
[{0}, {1}, {2}, {3}, {0, 1}, {0, 1}, {0, 1}]
[{0}, {1}, {2}, {3}, {0, 1, 2, 3}, {0, 1, 2, 3}, {0, 1, 2, 3}]
```
It seems to me that this is some kind of reference problem happening here

Reply:
Ugh, I forgot that I was in reviewer's mode, all my messages were pending.